codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ClientGenerator.java

```diff
@@ -63,41 +57,10 @@ public ClientGenerator(GenerationContext context) {
         this.model = context.model();
         this.writer = new RubyCodeWriter();
         this.rbsWriter = new RubyCodeWriter();
-
-        // Register all middleware
-        this.middlewareBuilder = new MiddlewareBuilder();
-        middlewareBuilder.addDefaultMiddleware(context);
-
-        context.integrations().forEach((integration) -> {
-            integration.modifyClientMiddleware(middlewareBuilder, context);
-        });
-
-        context.protocolGenerator().ifPresent((g) -> g.modifyClientMiddleware(middlewareBuilder, context));
-
-        // get all config
-        Set<ClientConfig> unorderedConfig = ClientConfig.defaultConfig();
-        unorderedConfig
-                .addAll(context.applicationTransport().getClientConfig());
-        unorderedConfig.addAll(middlewareBuilder.getClientConfig(context));
-        unorderedConfig.addAll(context.integrations()
-                .stream()
-                .map((i) -> i.getAdditionalClientConfig(context))
-                .flatMap(Collection::stream)
-                .collect(Collectors.toList())
-        );
-        context.protocolGenerator().ifPresent((g) -> unorderedConfig.addAll(g.getAdditionalClientConfig(context)));
-
-        clientConfig = unorderedConfig.stream()
-                .sorted(Comparator.comparing(ClientConfig::getName))
-                .collect(Collectors.toList());
-
-        LOGGER.fine("Client config: "
-                + clientConfig.stream().map((m) -> m.toString()).collect(Collectors.joining(",")));
-
         this.symbolProvider = new RubySymbolProvider(model, context.settings(), "Client", false);
     }
 
-    public void render() {
+    public void render(MiddlewareBuilder middlewareBuilder) {
```

Comment:
I'd still lean towards this being added to the constructor.  It also keeps us from having to pass it around to all of the individual methods.  Would we ever call render multiple times for the same Generator w/ different middleware? I don't think we would.

Reply:
Maybe we should stick it on context then